### PR TITLE
v1.10.14

### PR DESCRIPTION
**Pull request by StephenButtolph**

### CODE REVIEW

File: snow/networking/handler/handler.go [hunk 1]
```diff
@@ -215,27 +216,29 @@ func (h *handler) selectStartingGear(ctx context.Context) (common.Engine, error)
...
+	return engines.StateSyncer, engines.Bootstrapper.Clear(ctx)
 }
 
 func (h *handler) Start(ctx context.Context, recoverPanic bool) {
-	h.ctx.Lock.Lock()
-	defer h.ctx.Lock.Unlock()
-
 	gear, err := h.selectStartingGear(ctx)
 	if err != nil {
+		h.ctx.Lock.Unlock()
```
Comment (patrick-ogrady) [inline snow/networking/handler/handler.go:225]: nit: would be nice if there was a comment stating that `Start` assumes `h.ctx.Lock` is held now

File: snow/validators/manager.go [hunk 2]
```diff
@@ -49,27 +105,176 @@ type manager struct {
...
+	}
+
+	m.lock.RLock()
+	set, exists := m.subnetToVdrs[subnetID]
+	m.lock.RUnlock()
+	if !exists {
+		return nil, ErrMissingValidators
+	}
+
+	return set.Sample(size)
```
Comment (patrick-ogrady) [inline snow/validators/manager.go:253]: Unless I'm missing something, it doesn't seem safe to call `Sample` on a set without holding a lock: https://github.com/ava-labs/avalanchego/blob/2d8222da82114088a2a99d8d69f40846f1ea6208/utils/set/sampleable_set.go#L138-L151

Comment (patrick-ogrady) [reply snow/validators/manager.go:253]: Diregard, I see this is wrapped by `*vdrSet`

File: vms/platformvm/txs/txheap/heap.go [hunk 3]
```diff
@@ -23,107 +20,57 @@ type Heap interface {
...
+	h.heap.Push(txID, htx)
 }
 
 func (h *txHeap) Get(txID ids.ID) *txs.Tx {
-	index, exists := h.txIDToIndex[txID]
-	if !exists {
-		return nil
-	}
-	return h.txs[index].tx
+	got, _ := h.heap.Get(txID)
```
Comment (patrick-ogrady) [inline vms/platformvm/txs/txheap/heap.go:46]: I don't think `got` is guaranteed to be non-nil?

File: vms/platformvm/txs/txheap/heap.go [hunk 4]
```diff
@@ -23,107 +20,57 @@ type Heap interface {
...
 	return res
 }
 
 func (h *txHeap) Remove(txID ids.ID) *txs.Tx {
-	index, exists := h.txIDToIndex[txID]
-	if !exists {
-		return nil
-	}
-	return heap.Remove(h.self, index).(*txs.Tx)
+	removed, _ := h.heap.Remove(txID)
```
Comment (patrick-ogrady) [inline vms/platformvm/txs/txheap/heap.go:60]: I don't think `removed` is guaranteed to be non-nil?

File: vms/platformvm/txs/txheap/heap.go (hunk 3 above)
Comment (StephenButtolph) [reply vms/platformvm/txs/txheap/heap.go:46]: `got` is a struct, not a pointer. So it's `zero` value will produce the expected result here.

File: vms/platformvm/txs/txheap/heap.go (hunk 4 above)
Comment (StephenButtolph) [reply vms/platformvm/txs/txheap/heap.go:60]: `removed ` is a struct, not a pointer. So its `zero` value will produce the expected result here.

**Pull request merged**